src/Pages/MainPage/Containers/MainContainer.tsx

```diff
@@ -1,34 +1,44 @@
-import { useState } from "react";
-
+import { useState, useEffect } from "react";
 import { Task } from "../../../types/Task";
 import TaskForm from "../Components/TaskForm";
 import TaskList from "../Components/TaskList";
 
 const MainContainer: React.FC = () => {
-  const [tasks, setTasks] = useState<Task[]>([]);
+  const [tasks, setTasks] = useState<Task[]>(() => {
+    const savedTasks = localStorage.getItem("my-tasks");
+    if (savedTasks) {
+      return JSON.parse(savedTasks).map((task: Task) => ({
+        ...task,
+        dueDate: new Date(task.dueDate),
+      }));
+    }
+    return [];
+  });
```

Comment:
```suggestion
  const [tasks, setTasks] = 
    useState<Task[]>(JSON.parse(localStorage.getItem("my-tasks")) || [])
```

Can't we do something like this? Looks cleaner in my opinion.

Reply:
That was my initial approach, but dueDate throws an error in that approach I think. It's stored as strings in local storage, when I parse it back, it does not automatically get converted to a date object. Hence, had to do it as an extra step during parsing.